src/main/java/spoon/MavenLauncher.java

```diff
@@ -407,6 +411,139 @@ private String extractVariable(String value) {
 			return value;
 		}
 
+		private List<Version> getVersionsFromM2(String groupId, String artifactId) {
+			String groupIdPath = groupId.replace(".", "/");
+			File[] fileVersion = Paths.get(m2RepositoryPath, groupIdPath, artifactId).toFile().listFiles(file -> {
+				if (!file.isDirectory()) {
+					return false;
+				}
+				String version = file.getName();
+				return Paths.get(m2RepositoryPath, groupIdPath, artifactId, version, artifactId + "-" + version + ".jar").toFile().exists();
+			});
+			if (fileVersion == null) {
+				return Collections.emptyList();
+			}
+			List<Version> versions = Arrays.stream(fileVersion).map(f -> new Version(f.getName())).collect(Collectors.toList());
+			versions.sort(Comparator.reverseOrder());
+			return versions;
+		}
+
+		class Version implements Comparable<Version> {
```

Comment:
What about extracting the inner classes: the MavenLauncher seems now composed almost entirely of InheritanceModel. It seems easier to me to maintain if it's extracted.

Reply:
I don't think it is a good idea InheritanceModel, is completely useless outside MavenLauncher.
I agree that the architecture is not really nice but it is not the purpose of this PR.